Lib/python/pyrun.swg

```diff
@@ -1256,7 +1291,11 @@ SWIG_Python_SetSwigThis(PyObject *inst, PyObject *swig_this)
       dict = PyDict_New();
       *dictptr = dict;
     }
-    return PyDict_SetItem(dict, SWIG_This(), swig_this);
+    if (dict) {
```

Comment:
Further OOM fix: `PyDict_New` may return `NULL`, in which case `PyDict_SetItem` should not be called with a `NULL` dictionary (it will be dereferenced in `PyDict_Check(op)` & crash).